src/serviceClients/catalogClient.ts

```diff
@@ -49,6 +49,23 @@ export class CatalogClient extends HttpClient {
     return res[0];
   }
 
+  @withSpanAsyncV4
+  public async findRecordById(identifier: string): Promise<IFindResponseRecord | undefined> {
```

Comment:
why dont you reuse the method above and add paramter of id?
`public async findRecord(productId: string, productVersion?: string, productType?: string): Promise<IFindResponseRecord | undefined> {`